### DISCUSSION TITLE
the diffrrence of elemental solution and nadal solution

### DISCUSSION THREAD

**Discussion by xiaojbing** [idaholab/moose#21985] [Q&A General] [answered]
Hello,
In moose, the elemental solution is the variable value at quadrature point. and then nadal solution is computed by values at quadrature points.
I want to know what is is the specific process that get nodal solution from elemental solution in moose.
I heard in ansys, every element will have nodal solution, which is computed from variable value at quadrature point in this element. When a node is shared by several elements, the final nodal solution is the average of nodal solution of respective element.

**Comment by YaqiWang** [comment]
I am not aware of a nodal auxiliary kernel that can take an elemental variable, and perform the average (possibly should be weighted by the angle/solid angle of the element contribution on the node). MOOSE output system can output elemental variables into nodal variables for visualization. I did not check the code but assume that is using libMesh functionalities and is close to what you want.

**Comment by GiudGiud** [comment, marked as answer]
For an elemental variable:
- If the node is a quadrature point, and if the FE family is continuous, then the nodal value is the quadrature point value of the variable
- if the node is a Qp and the FE family is discontinuous, the nodal value is ill-defined, you'd need to specify which element to use
- if the node is not a Qp, and the FE family is continuous, the nodal value may be computed by evaluating the shape functions there
- if the node is not a Qp and the FE family is discontinuous, the nodal value is ill-defined, you d need to specify from which element to come from when evaluating

You can indeed do projections from one type to another using either the Transfers system, and in some cases the SelfAux
